marconi-chain-index/src/Marconi/ChainIndex/Indexers/EpochState.hs

```diff
@@ -435,20 +444,22 @@ instance Buffered EpochStateHandle where
                       $ fmap epochStateEventChainTip nonEmptyEvents
 
               ledgerStateFilePaths <-
-                  mapMaybe (\fp -> fmap (fp,) $ chainTipsFromLedgerStateFilePath fp)
+                  mapMaybe (\fp -> fmap (fp,) $ readLedgerStateFileMetadata fp)
                   <$> listDirectory ledgerStateDirPath
 
               -- Delete volatile LedgerState which have become immutable.
               let oldVolatileLedgerStateFilePaths =
                       fmap fst
-                      $ filter (\(_, (isVolatile, _, _, blockNo)) ->
-                          isVolatile && not (isBlockRollbackable securityParam blockNo chainTip))
+                      $ filter (\(_, LedgerStateFileMetadata { lsfMetaIsVolatile, lsfMetaBlockNo }) ->
```

Comment:
In this entire delete section, if the goal is to keep just the latest non-volatile ledger state, then whether a ledger state was volatile or not at the time it was written is now outdated and can be ignored (?) -- just filter with `isBlockRollbackable` to remove the ledger states which are volatile _now_, and from the remaining list remove all the older ledger states.

Reply:
> if the goal is to keep just the latest non-volatile ledger state

No, the goal is to keep the latest non-volatile LedgerState *and* all volatile LedgerStates.

> just filter with isBlockRollbackable to remove the ledger states which are volatile now,

But I don't want to filter LedgerStates which are volatile now, I need to keep them in case of a rollback.

However, your point does bring up that something is maybe wrong here... 
In the first deletion part, I'm removing all volatile LedgerStates that have become immutable. However, that's seems wrong AFAIU. I need to *rename* the file from `ledgerState_volatile_...` to `ledgerState_...` and then the next delete part will delete all LedgerStates except the latest one.

Does that make this more valid?

Reply:
Yup, this now makes it clearer! 

What I was trying to get at at the other comment is to not add the `_volatile` tag at all, because whether the ledger state is volatile or not depends on where the current tip is when the program is using the file (the _now_ for the program), and not where the current tip was when the file was written. I.e maybe on of the volatile ledger states that now has become immutable (and is deleted) is actually the one that got written just a few lines above. So it's possible to save a write/delete cycle.

Reply:
Alright that makes sense. I'll do the following changes:

* remove the "volatile" keyboard in the filename
* modify the deletion logic to only keep the latest immutable LedgerStates along with *all* volatile LedgerStates, and assess if a LedgerState is immutable using the available information

Reply:
@eyeinsky I applied the suggestions here 479869bf8868db054f4c794b413eea492465fbaa
Can you validate?

Reply:
Still can't the volatility of a ledger state be determined dynamically, from the "live" chain tip be gotten from the latest block that came in over chainsync instead? But I see now that in [resumeFromStorage](https://github.com/input-output-hk/marconi/pull/37/files#diff-50d35d89163b61b8994cdf3b2e505d43ec2f833875c43469c9c075053ebfdfa1R609-R654) the actual tip at the time of resuming isn't passed in, so with the current indexer API I guess this is the best we can do.

Reply:
That's right, it's unfortunately not possible to have it in `resumeFromStorage`. That is, it is possible to query the tip to the local node *before* running the indexers, and then pass that info in `EpochSPDHandle`. But then, we would have an information in `EpochSPDHandle` that will *only* be used for resuming and will be obsolete afterwards. Not sure if that's worth it. Thoughts?